gnovm/pkg/gnolang/op_expressions.go

```diff
@@ -796,6 +796,6 @@ func (m *Machine) doOpFuncLit() {
 func (m *Machine) doOpConvert() {
 	xv := m.PopValue()
 	t := m.PopValue().GetType()
-	ConvertTo(m.Alloc, m.Store, xv, t)
+	ConvertTo(m.Alloc, m.Store, xv, t, false)
```

Comment:
Here, we don't know if `xv` is a constant or not. Forcing `false` for `isConst` parameter here is incorrect for constant expressions. For example, the following sample should fail but it doesn't:

```go
println(int(float64(1 + 0.5)))
```

This case seems hard to fix with the current design of constant processing.

Reply:
This does fail

Reply:
ok